starport-core/src/main/scala/com/krux/starport/metric/CloudWatchReporter.scala

```diff
@@ -0,0 +1,886 @@
+package com.krux.starport.metric
+
+import java.util._
+import java.util.concurrent.{ConcurrentHashMap, Future, TimeUnit}
+
+import scala.collection.JavaConverters._
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsync
+import com.amazonaws.services.cloudwatch.model.{
+  Dimension,
+  MetricDatum,
+  PutMetricDataRequest,
+  PutMetricDataResult,
+  StandardUnit,
+  StatisticSet
+}
+import com.codahale.metrics.{
+  Clock,
+  Counter,
+  Counting,
+  Gauge,
+  Histogram,
+  Meter,
+  Metered,
+  MetricFilter,
+  MetricRegistry,
+  ScheduledReporter,
+  Snapshot,
+  Timer
+}
+import org.slf4j.{Logger, LoggerFactory}
+import com.krux.starport.metric.CloudWatchReporter._
+
+/**
+ * Reports metrics to <a href="http://aws.amazon.com/cloudwatch/">Amazon's CloudWatch</a> periodically.
+ * <p>
+ * Use {@link CloudWatchReporter.Builder} to construct instances of this class. The {@link CloudWatchReporter.Builder}
+ * allows to configure what aggregated metrics will be reported as a single {@link MetricDatum} to CloudWatch.
+ * <p>
+ * There are a bunch of {@code with*} methods that provide a sufficient fine-grained control over what metrics
+ * should be reported
+ */
+class CloudWatchReporter private (builder: Builder)
+    extends ScheduledReporter(
+      builder.metricRegistry,
+      "coda-hale-metrics-cloud-watch-reporter",
+      builder.metricFilter,
+      builder.rateUnit,
+      builder.durationUnit
+    ) {
+
+  /**
+   * We only submit the difference in counters since the last submission. This way we don't have to reset
+   * the counters within this application.
+   */
+  private val lastPolledCounts: Map[Counting, Long] = new ConcurrentHashMap()
+
+  private val namespace: String = builder.namespace
+
+  private val cloudWatchAsyncClient: AmazonCloudWatchAsync = builder.cloudWatchAsyncClient
+
+  private val rateUnit: StandardUnit = builder.cwRateUnit
+
+  private val durationUnit: StandardUnit = builder.cwDurationUnit
+
+  private val highResolution: Boolean = builder.highResolution
+
+  override def report(
+    gauges: java.util.SortedMap[String, Gauge[_]],
+    counters: java.util.SortedMap[String, Counter],
+    histograms: java.util.SortedMap[String, Histogram],
+    meters: java.util.SortedMap[String, Meter],
+    timers: java.util.SortedMap[String, Timer]
+  ): Unit = {
+
+    if (builder.dryRun) {
```

Comment:
let's remove the dryRun mode for simplicity

Reply:
OK.